### PR TITLE
Phase2-hgx316A Second step to enable a scenario with a HGCal geometry having no cell definition in the G4 geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Second step to enable a scenario with a HGCal geometry having no cell definition in the G4 geometry

#### PR validation:

Use test workflows in Geometry/HGCaCommonData/test/python

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , as I understand there are two approaches in HGCal geometry : cell in a layer is defined or not. The second means that cell number should be restored using coordinates of the hit. Why it is called "weighted" cell?

Inside SD code the name of geometry configuration and SD name are used to find detector ID. May be more optimal to use enumerators?

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@srimanob Please approve this PR

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -1338,16 +1338,16 @@ void HGCalDDDConstants::waferFromPosition(const double x,
       if (ktr != hgpar_->waferInfoMap_.end()) {
         place = HGCalCell::cellPlacementIndex(1, HGCalTypes::layerFrontBack(layertype), (ktr->second).orient);
         part = (ktr->second).part;
+#ifdef EDM_ML_DEBUG
         if (debug)
```
Comment (srimanob) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:1342]: Why do you still need this `if`?

Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:1342]: This is a time-critical code. So avoid the testing for standard processing - and once compiled, need to depend on the debug flag

**Comment by srimanob** [comment]
+Upgrade

**Comment by perrotta** [comment]
+1

**Pull request merged**